src/utils/signing-nonce.ts

```diff
@@ -0,0 +1,32 @@
+import { randomStringForEntropy } from '@stablelib/random';
+import { validateAddress, ValidationResult } from '@taquito/utils';
+import memoizee from 'memoizee';
+
+import { CodedError } from './errors';
+
+const SIGNING_NONCE_TTL = 5 * 60_000;
+
+export const getSigningNonce = memoizee(
+  (pkh: string) => {
+    if (validateAddress(pkh) !== ValidationResult.VALID) throw new CodedError(400, 'Invalid address');
+
+    return buildNonce();
+  },
+  {
+    max: 500,
```

Comment:
This approach may fail if we get too many users simultaneously completing the quest. Please consider using the database